python/CommonFramework.h

```diff
@@ -15,20 +15,43 @@ class PythonBase {
 public:
     /** \name ctor */ //@{
     PythonBase();
+    virtual ~PythonBase();
     //@}
 
+    /**
+       Handles boost::python::error_already_set.
+       If the error is SystemExit the python interpreter is finalized
+       and no longer available.
+
+       Call PyErr_Print() if the exception is an error.
+
+       HandleErrorAlreadySet is idempotent, calling it mutliple times
+       won't crash or hang the process.
+     */
+    void         HandleErrorAlreadySet();
+    /**
+       IsPythonRunning returns true is the python interpreter is
+       initialized.  It is typically called after
+       HandleErrorAlreadySet() to determine if the error caused the
+       interpreter to shutdown.
+     */
+    bool         IsPythonRunning();
+
     bool         Initialize();                         // initializes and runs the Python interpreter, prepares the Python environment
     void         Finalize();                           // stops Python interpreter and releases its resources
     virtual bool InitModules() = 0;                    // initializes Python modules, must be implemented by derived classes
-    bool         ExecScript(const std::string script); // executes a Python script
-    bool         SetCurrentDir(const std::string dir); // sets Python current work directory
-    bool         AddToSysPath(const std::string dir);  // adds directory to Python sys.patch
+    void         ExecScript(const std::string script); // executes a Python script or throws error_already_set
+    void         SetCurrentDir(const std::string dir); // sets Python current work directory or throws error_already_set
+    void         AddToSysPath(const std::string dir);  // adds directory to Python sys.path or throws error_already_set
     void         SetErrorModule(boost::python::object& module); // sets Python module that contains error report function defined on the Python side
     std::vector<std::string> ErrorReport();            // wraps call to error report function defined on the Python side
 
 private:
+    // A copy of the systemExit exception to compare with returned
+    // exceptions.  It can't be created in the exception handler.
+    boost::python::object systemExit;
```

Comment:
I know it's just a helper object, but IMO it should still conform to standard naming style: `m_system_exit`.